lively.ide/components/editor.js

```diff
@@ -144,17 +146,20 @@ export class InteractiveComponentDescriptor extends ComponentDescriptor {
   async ensureComponentDefBackup () {
     if (this._backupComponentDef) return;
     const { moduleId, exportedName } = this[metaSymbol];
-    const source = await module(moduleId).source();
-    const { start, end } = findComponentDef(await module(moduleId).ast(), exportedName);
+    const source = await this.targetModule.source();
+    const { start, end } = findComponentDef(await this.targetModule.ast(), exportedName);
     this._backupComponentDef = source.slice(start, end);
   }
 
   /**
    * Revert the component definition back to when we started the edit session.
    */
   async reset () {
-    await replaceComponentDefinition(this._backupComponentDef, this.componentName, this.moduleName);
+    const dependants = this.stylePolicy._dependants;
+    await replaceComponentDefinition(this._backupComponentDef, this.componentName, this.targetModule);
     this._dirty = false;
+    this.stylePolicy._dependants = dependants;
+    // FIXME: carry over dependents for now
```

Comment:
Why is this necessary?

Reply:
Dependents includes not only other components but also policies that are installed in morph instances. Unlike components, which are defined in modules and re-evaluate themselves, these instance policies can only be recorded at the point when they are initialized. So in order not to loose them, we need to "carry over" the dependents array.

Reply:
I might be missing something obvious here, but I do not see how `replaceComponentDefinition` changes the `stylePolicy`? And this is the case where we would need to keep the `dependants` values around, no?

Reply:
`replaceComponentDefinition()` in this case reverts the souce code AND also reloads the module which causes the `stylePolicy` to be changed after it finishes execution. The `stylePolicy` before and after are different objects.

Reply:
Ah, got it!